src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -419,7 +424,7 @@ private static bool IsHostRelative(string path)
             }
 
             var loginUri = Options.LoginPath + QueryString.Create(Options.ReturnUrlParameter, redirectUri);
-            var redirectContext = new CookieRedirectContext(Context, Scheme, Options, BuildRedirectUri(loginUri), properties);
+            var redirectContext = new CookieRedirectContext(Context, Scheme, Options, properties, BuildRedirectUri(loginUri));
```

Comment:
It's for consistency with the other events, where `AuthenticationProperties` generally precedes parameters that are specific to the context class (e.g `OAuthRedirectToAuthorizationContext`).

That said, if you don't like it, feel free to revert it.